iceoryx_posh/include/iceoryx_posh/version/version_info.hpp

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#ifndef IOX_POSH_VERSION_VERSION_INFO_HPP
+#define IOX_POSH_VERSION_VERSION_INFO_HPP
+
+#include "iceoryx_posh/version/compatibility_check_level.hpp"
+#include "iceoryx_utils/cxx/serialization.hpp"
+#include "iceoryx_utils/fixed_string/string100.hpp"
+#include "iceoryx_versions.hpp"
+
+#include <cstdint>
+
+namespace iox
+{
+namespace version
+{
+template <size_t SizeValue>
```

Comment:
This is great! But I think we should move it into `cxx/helplets.hpp`. While doing that could you please use `uint64_t` instead of `size_t`. The reason is that `size_t` is sometimes 32-bit and sometimes 64-bit in size and on some architectures it is even not unsigned (like Mac OS).

Reply:
done